packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -258,7 +258,13 @@ export class PendingStateManager implements IDisposable {
         }
 
         if (local) {
-            return { localAck: false, localOpMetadata: this.processPendingLocalMessage(message) };
+            if (this.firstStashedCSN !== undefined &&
+                message.clientSequenceNumber < this.firstStashedCSN &&
+                this.previousClientIds.has(message.clientId)) {
+                // this is an op from a previous container that was already ACKed before it was closed
+                return { localAck: false, localOpMetadata: undefined };
```

Comment:
I do not remember where we landed RE processing multiple disconnects (and thus multiple clientIds), but given that clientSequenceNumber is reset any time clientId changes, I'd think that this.previousClientIds can not be an array, it needs to be just a string (one value). Basically, switch of clientIds (and recording into this.previousClientIds) should happen only when we get to "connected" state, which should mean that we know for sure ops with old clientId can no longer arrive as we observed them all, and thus there is no need to track old clientsIds.